sklearn/utils/estimator_checks.py

```diff
@@ -1027,6 +1026,9 @@ def _check_transformer(name, transformer_orig, X, y):
     else:
         # check for consistent n_samples
         assert X_pred.shape[0] == n_samples
+        if transformer_clone.n_features_out_ is not None:
```

Comment:
Most of the transformers at this stage don't have the `n_features_out_` as an attribute, do they?

Reply:
what do you mean by "at this stage"? This PR implements it for all estimators but ``FunctionTransformer``, which sets it explicitly to ``None``. ``fit`` is called a couple lines above, so they should all have it here.